community-build/test/java/dotty/communitybuild/FailureSummarizer.java

```diff
@@ -0,0 +1,36 @@
+package dotty.communitybuild;
+
+import java.util.List;
+
+import org.junit.runner.Description;
+import org.junit.runner.Result;
+import org.junit.runner.notification.Failure;
+import org.junit.runner.notification.RunListener;
+
+public class FailureSummarizer extends RunListener {
+  @Override
+  public void testRunFinished(Result result) throws Exception {
+    super.testRunFinished(result);
+    if (result.getFailureCount() > 0) {
+      Thread.sleep(500);  // pause to give sbt log buffers some time to flush
```

Comment:
Is that still needed if you output with System.out instead of System.err ?

Reply:
I was seeing interspersed and/or too early output without it, no matter which file handle I chose.  It's a bit hacky, but I don't think there's a way here to route the output via the sbt logger?

sbt/junit-interface has some functionality in current master (option --summary=2) that might also work to solve this issue, but there doesn't seem to have been a release since 0.11 over 6 years ago.

Reply:
Interesting, I think we should try to push for a new release of junit-interface.

Reply:
https://github.com/sbt/junit-interface/issues/94. Meanwhile this PR is already a great improvement even if we have to rely on some hacks so I'd be happy to get it in.
Another approach we might to explore would be using the github annotation API to get info from failed tests reported back in the github UI, there's some github actions to do this based on the XML reports that JUnit generates but I haven't tried them: https://github.com/marketplace/actions/junit-report-to-annotations, https://github.com/marketplace/actions/junit-report

Reply:
The annotation API is an interesting avenue for exploration; I'll take a look and see how we might make use of it, not just in regards to the community build but for all the JUnit tests.  But that will likely take time and experimentation and this PR provides a quick (albeit hacky) fix in the interim.